### ISSUE TITLE
Roaring64Bitmap toArray() function throw exception java.lang.ArrayIndexOutOfBoundsException。

### ISSUE THREAD

**Issue by caoxiaoyuan7242** [RoaringBitmap/RoaringBitmap#420]
**Describe the bug**
Roaring64Bitmap toArray() function throw exception java.lang.ArrayIndexOutOfBoundsException

**To Reproduce**
```Java
        Random random = new Random();
        Roaring64Bitmap bitmap = new Roaring64Bitmap();
        for (long i = 0; i < 20000; i++) {
            bitmap.addLong(random.nextInt(10000000));
        }

        Roaring64Bitmap bitmap2 = new Roaring64Bitmap();
        for (long i = 1; i < 40000; i++) {
            int value = random.nextInt(10000000);
            bitmap2.addLong(value);
        }

        //bit and
        bitmap.and(bitmap2);

        //to array
        bitmap.toArray();
```

**Java version:*jdk 1.8*
RoaringBitmap version:0.9.0

Please tell us which Java version you are using.

**Comment by lemire** [comment]
@weijietong Would yo have a look at the issue?

**Comment by lemire** [comment]
I am marking this as a potential bug until someone can verify it. Thanks for your report.

**Comment by caoxiaoyuan7242** [comment]
i find the bug

  protected LongIterator toIterator(final LeafNodeIterator keyIte, boolean reverse) {
    return new LongIterator() {
      private byte[] high;
      private CharIterator charIterator;
      private boolean hasNextCalled = false;

      @Override
      public boolean hasNext() {
        hasNextCalled = true;
        if (charIterator != null && !charIterator.hasNext()) {
          if (keyIte.hasNext()) {
            LeafNode leafNode = keyIte.next();
            high = leafNode.getKeyBytes();
            long containerIdx = leafNode.getContainerIdx();
            Container container = highLowContainer.getContainer(containerIdx);
            if (!reverse) {
              charIterator = container.getCharIterator();
            } else {
              charIterator = container.getReverseCharIterator();
            }
            return true;
          } else {
            return false;
          }
        }
        if (charIterator != null && charIterator.hasNext()) {
          return true;
        }
        if (charIterator == null) {
          if (keyIte.hasNext()) {
            LeafNode leafNode = keyIte.next();
            high = leafNode.getKeyBytes();
            long containerIdx = leafNode.getContainerIdx();
            Container container = highLowContainer.getContainer(containerIdx);
            if (!reverse) {
            **_//here charIterator can be has no next_**
              charIterator = container.getCharIterator();
            } else {
              charIterator = container.getReverseCharIterator();
            }
            return true;
          } else {
            return false;
          }
        }
        return false;
      }

      @Override
      public long next() {
        boolean hasNext = true;
        if (!hasNextCalled) {
          hasNext = hasNext();
          hasNextCalled = false;
        }
        if (hasNext) {
         **//here charIterator can be has no _next**_
          char low = charIterator.next();
          return LongUtils.toLong(high, low);
        } else {
          throw new IllegalStateException("empty");
        }
      }

      @Override
      public LongIterator clone() {
        throw new UnsupportedOperationException("TODO");
      }
    };
  }

**Comment by lemire** [comment]
@caoxiaoyuan7242 Would you be willing to prepare a pull request that includes a fix as well as a test for the issue you have encountered?

**Comment by caoxiaoyuan7242** [comment]
> @caoxiaoyuan7242 Would you be willing to prepare a pull request that includes a fix as well as a test for the issue you have encountered?

ok，I'll submit a  pull request  tomorrow. Now it's twelve o'clock in the evening

**Comment by lemire** [comment]
Thank you.

**Comment by weijietong** [comment]
Sorry for the late response. Thanks for your report @caoxiaoyuan7242 , your analysis about the bug is right.  According to the current implementation, the stored value containers in the `HighLowContainer` maybe empty ones. The `LongIterator` does not check that case, so that case cause this exception. Wait for your PR.

**Comment by caoxiaoyuan7242** [comment]
@weijietong @lemire hi,i submit a pull request for this bug.

**Comment by caoxiaoyuan7242** [comment]
I would like to ask when the official version of the package will be released. We need to use the repaired functions

**Comment by weijietong** [comment]
The corresponding PR has been merged into the master, you can get a fresh version from the master. The official version will be released according to @lemire 's decision.

**Comment by lemire** [comment]
> The official version will be released according to @lemire 's decision.

I will issue a patch release today. Obviously we want people to have access to this fix.

**Comment by caoxiaoyuan7242** [comment]
> > The official version will be released according to @lemire 's decision.
> 
> I will issue a patch release today. Obviously we want people to have access to this fix.

That's great. Thank you. It would be better if you could tell me the new version number

**Comment by lemire** [comment]
Fixed in version 0.9.3.

I have not checked that this version has propagated to maven central. Be mindful that it is not a fast process.
